### PR TITLE
private contributions, add profile_id to contributions, add SoulKeyActivityPage

### PR DESCRIPTION
**Pull request by CryptoGraffe**
Adds: private contributions that are only visible to key_holders. Adds private_stream_visibility table to cache the permissions for this. Lots of styling.

TODO:

- [x] editing contributions w/o circle_id doesn't work
- [x] make sure all the other contribution / activity stuff works
- [x] add permissions for soulkey holders to see private_stream stuff
- [x] figure out how to show activity for just me vs everyone else
- [x] loader thing after submit never dissipates
- [x] placeholder text and info helper text is "team" centric
- [x] activity avatar should link to soulkey page when in soulkey view
- [x] error when reacting , prob permission based {
    "errors": [
        {
            "message": "check constraint of an insert/update permission has failed",
            "extensions": {
                "path": "$.selectionSet.insert_reactions_one.args.object",
                "code": "permission-error"
            }
        }
    ]
}

### <samp>🤖 Generated by Copilot at a386a12</samp>

No summary available (Limit exceeded: required to process 91558 tokens, but only 50000 are allowed per call)

### <samp>🤖 Generated by Copilot at e2266e1</samp>

> _`Private_stream` of blood and tears_
> _No `circle_id` to ease your fears_
> _You `insertContributionActivity` in the dark_
> _But the `profile_id` will leave a mark_

### <samp>🤖 Generated by Copilot at a386a12</samp>

No walkthrough available (Limit exceeded: required to process 91558 tokens, but only 50000 are allowed per call)

### CODE REVIEW

File: src/features/soulkeys/api/updateHolders.ts [hunk 1]
```diff
@@ -191,23 +193,175 @@ async function updateKeyHoldersTable(holdersToUpdate: InsertOrUpdateHolder[]) {
...
+  }
+
+  await deleteFromKeysHolderCacheAndPrivateVisibility(deleteHolders);
+  // for each holder_pair, make sure they have a private_stream_visibility row
+
+  /*
+  for each of the unique holders, update their private_stream_visibility table
+   */
+  // for (const holder of uniqueHolders) {
+  //   // for each holder, insert into private_stream_visibility for subject,address and address,subject
```
Comment (crabsinger) [inline src/features/soulkeys/api/updateHolders.ts:253]: maybe make it an event trigger

File: src/features/activities/useInfiniteActivities.tsx [hunk 2]
```diff
@@ -110,14 +110,21 @@ export const useInfiniteActivities = (
 
 export type Activity = Awaited<ReturnType<typeof getActivities>>[number];
 
-export type Contribution = Activity &
-  Required<Pick<Activity, 'contribution' | 'actor_profile' | 'circle'>>;
+type c1 = Activity &
```
Comment (crabsinger) [inline src/features/activities/useInfiniteActivities.tsx:113]: cleanup

File: src/features/activities/DeletedRow.tsx [hunk 3]
```diff
@@ -1,38 +1,48 @@
...
           css={{ cursor: 'auto' }}
         />
-      </Flex>
-    </Flex>
+      </div>
+    </div>
   );
+
+  // <Flex alignItems="center">
+  //   {activity.actor_profile_public && (
```
Comment (crabsinger) [inline src/features/activities/DeletedRow.tsx:25]: cleanup

File: src/features/activities/ContributionRow.tsx [hunk 4]
```diff
@@ -90,13 +91,15 @@ export const ContributionRow = ({
...
-                  contributionId={activity.contribution.id}
-                  circleId={activity.circle.id}
-                />
+                <>
+                  <ContributionForm
+                    css={{ textarea: { background: '$surfaceNested ' } }}
+                    description={activity.contribution.description}
+                    setEditingContribution={setEditingContribution}
+                    contributionId={activity.contribution.id}
+                    // circleId={activity.circle ? activity.circle.id : undefined}
```
Comment (crabsinger) [inline src/features/activities/ContributionRow.tsx:100]: remove

File: src/features/soulkeys/api/updateHolders.ts (hunk 1 above)
Comment (CryptoGraffe) [reply src/features/soulkeys/api/updateHolders.ts:253]: added todo and deleted these comments

File: src/features/activities/DeletedRow.tsx (hunk 3 above)
Comment (CryptoGraffe) [reply src/features/activities/DeletedRow.tsx:25]: We might want this stuff to debug

File: src/features/activities/ContributionRow.tsx (hunk 4 above)
Comment (CryptoGraffe) [reply src/features/activities/ContributionRow.tsx:100]: undeleted this cuz its actually good

File: src/features/activities/useInfiniteActivities.tsx (hunk 2 above)
Comment (CryptoGraffe) [reply src/features/activities/useInfiniteActivities.tsx:113]: done

**Review by crabsinger — APPROVED** [review]

**Pull request merged**